### ISSUE TITLE
ReverseProxyResource adds an extra '/' which confuses the backend web server

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2905]
|<img alt="trohed's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| trohed reported|
|-|-|
|Trac ID|trac#2905|
|Type|defect|
|Created|2007-11-16 17:12:46Z|

It looks like proxy.ReverseProxyResource adds an extra '/' which confuses the backend web server.  This appears to only effect the backend web server if its serving up static files.

I've attached a possible fix and a test suite.

Attachments:

* [reverse_proxy.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/faa/faa414d679fc9185effcc6e5122a67066657c338/ff42059d09fa46e67b570315ea527b6974f3dfc7.patch) (508 bytes) - added by trohed on 2007-11-16 17:14:04Z - 
* [test_proxy.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/faa/faa414d679fc9185effcc6e5122a67066657c338/02ee240cee1679557b9649530fcf1bad01816ffe.py) (4268 bytes) - added by trohed on 2007-11-16 17:14:36Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2905 2905
type__defect defect
reporter__trohed trohed
priority__normal normal
milestone__Web_0_7 Web-0.7
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__reverse_proxy reverse proxy
time__1195233166000000 1195233166000000
changetime__1195597424000000 1195597424000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

Thanks a lot for your contribution. The test suite should probably be added to [#11333](https://github.com/twisted/twisted/issues/11333) though.

The tests would be even better if they didn't involve socket connections. I don't know web.proxy much, but I guess it's possible.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

Okay, I may be wrong, but the only way that ```ReverseProxyResource``` get an extra slash is if you pass it to the constructor. That's probably only a documentation issue that will be solved by [#11333](https://github.com/twisted/twisted/issues/11333).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

(In [21905]) Add documentation.

Refs [#11333](https://github.com/twisted/twisted/issues/11333)
Refs [#2905](https://github.com/twisted/twisted/issues/2905)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set status to `closed`|
|-|-|

(In [21922]) Merge proxy-tests-238

Author: therve
Reviewers: exarkun, jml
Fixes [#11333](https://github.com/twisted/twisted/issues/11333)
Fixes [#2905](https://github.com/twisted/twisted/issues/2905)

Add tests to twisted.web.proxy.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
